src/vs/workbench/api/common/extHostEditorTabs.ts

```diff
@@ -8,43 +8,57 @@ import { IEditorTabDto, IExtHostEditorTabsShape } from 'vs/workbench/api/common/
 import { URI } from 'vs/base/common/uri';
 import { Emitter, Event } from 'vs/base/common/event';
 import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { ViewColumn } from 'vs/workbench/api/common/extHostTypes';
 
 export interface IEditorTab {
-	name: string;
-	group: number;
-	resource: vscode.Uri
+	label: string;
+	viewColumn: ViewColumn;
+	resource?: vscode.Uri
 	isActive: boolean
 }
 
 export interface IExtHostEditorTabs extends IExtHostEditorTabsShape {
 	readonly _serviceBrand: undefined;
 	tabs: readonly IEditorTab[];
-	onDidChangeTabs: Event<void>;
+	activeTab: IEditorTab | undefined;
+	onDidChangeActiveTab: Event<IEditorTab | undefined>;
+	onDidChangeTabs: Event<IEditorTab[]>;
 }
 
 export const IExtHostEditorTabs = createDecorator<IExtHostEditorTabs>('IExtHostEditorTabs');
 
 export class ExtHostEditorTabs implements IExtHostEditorTabs {
 	readonly _serviceBrand: undefined;
 
-	private readonly _onDidChangeTabs = new Emitter<void>();
-	readonly onDidChangeTabs: Event<void> = this._onDidChangeTabs.event;
+	private readonly _onDidChangeTabs = new Emitter<IEditorTab[]>();
+	readonly onDidChangeTabs: Event<IEditorTab[]> = this._onDidChangeTabs.event;
+
+	private readonly _onDidChangeActiveTab = new Emitter<IEditorTab | undefined>();
+	readonly onDidChangeActiveTab: Event<IEditorTab | undefined> = this._onDidChangeActiveTab.event;
 
 	private _tabs: IEditorTab[] = [];
+	private _activeTab: IEditorTab | undefined;
 
 	get tabs(): readonly IEditorTab[] {
 		return this._tabs;
 	}
 
+	get activeTab(): IEditorTab | undefined {
+		return this._activeTab;
+	}
+
 	$acceptEditorTabs(tabs: IEditorTabDto[]): void {
+		this._activeTab = undefined;
 		this._tabs = tabs.map(dto => {
-			return {
-				name: dto.name,
-				group: dto.group,
+			return Object.freeze({
+				label: dto.label,
+				viewColumn: dto.viewColumn,
 				resource: URI.revive(dto.resource),
 				isActive: dto.isActive
-			};
+			});
 		});
-		this._onDidChangeTabs.fire();
+		this._activeTab = this._tabs.find(t => t.isActive);
```

Comment:
nit but you already iterated over this array (via `map`) and could already know the index of the active tab. No need for a second iteration

Reply:
Thanks!